gourmet/gtk_extras/TextBufferMarkup.py

```diff
@@ -15,337 +15,123 @@
-                iter.set_offset(start_pos)
+            itr = self.get_iter_at_mark(self.insert_)
+            if itr.inside_word():
+                start_pos = itr.get_offset()
+                itr.forward_word_end()
+                word_end = itr.get_offset()
+                itr.backward_word_start()
+                word_start = itr.get_offset()
+                itr.set_offset(start_pos)
                 bounds = (self.get_iter_at_offset(word_start),
                           self.get_iter_at_offset(word_end+1))
             else:
-                bounds = (iter,self.get_iter_at_offset(iter.get_offset()+1))
+                bounds = (itr, self.get_iter_at_offset(itr.get_offset()+1))
         return bounds
 
-    def apply_tag_to_selection (self, tag):
-        selection = self.get_selection()
-        if selection:
-            self.apply_tag(tag,*selection)
-
-    def remove_tag_from_selection (self, tag):
-        selection = self.get_selection()
-        if selection:
-            self.remove_tag(tag,*selection)
-
-    def remove_all_tags (self):
-        selection = self.get_selection()
-        if selection:
-            for t in list(self.tags.values()):
-                self.remove_tag(t,*selection)
-
-class InteractivePangoBuffer (PangoBuffer):
-    def __init__ (self,
-                  normal_button=None,
-                  toggle_widget_alist=[]):
-        """An interactive interface to allow marking up a Gtk.TextBuffer.
-        txt is initial text, with markup.
-        buf is the Gtk.TextBuffer
-        normal_button is a widget whose clicked signal will make us normal
-        toggle_widget_alist is a list that looks like this:
-        [(widget, (font,attr)),
-         (widget2, (font,attr))]
-         """
-        PangoBuffer.__init__(self)
-        if normal_button: normal_button.connect('clicked',lambda *args: self.remove_all_tags())
+    def apply_tag_to_selection(self, tag):
+        start, stop = self.get_selection()
+        self.apply_tag(tag, start, stop)
+
+    def remove_tag_from_selection(self, tag):
+        start, stop = self.get_selection()
+        self.remove_tag(tag, start, stop)
+
+    def remove_all_tags(self):
+        start, stop = self.get_selection()
+        for tag in list(self.tags.values()):
+            self.remove_tag(tag, start, stop)
+
+
+class InteractivePangoBuffer(PangoBuffer):
+    def __init__(self,
+                 normal_button=None,
+                 toggle_widget_alist: Optional[List] = None):
+        super().__init__()
+
+        if normal_button is not None:
+            normal_button.connect('clicked', self.remove_all_tags)
         self.tag_widgets = {}
         self.internal_toggle = False
-        self.insert_ = self.get_insert()
-        self.connect('mark-set',self._mark_set_cb)
-        self.connect('changed',self._changed_cb)
-        for w,tup in toggle_widget_alist:
-            self.setup_widget(w,*tup)
+        self.insert_: Gtk.TextMark = self.get_insert()
```

Comment:
This could be `self._insert`: it's not used outside the class

Reply:
That would be better to decrease the visible interface